> @aambrosano on this
> 
> > For the record, OpenSSL::OpenSSL is not really the right target. FindOpenSSL will actually create OpenSSL::crypto and OpenSSL::ssl, however linking against those targets is still generating errors at the CMake generation step. Couldn't figure out why yet.
> 
> It doesn't work because the targets should be OpenSSL::Crypto and OpenSSL::SSL not OpenSSL::crypto and OpenSSL::ssl.

I was just typing targets names off the top of my head when commenting, but was using the right names when coding. That will not work anyway because for some reason targets found within a function are not available outside of the scope of the function. Trying to link against `OpenSSL::SSL` and `OpenSSL::Crypto` will result in an error message like this one:

```
CMake Error in CMakeLists.txt:
  The dependency target "OpenSSL::SSL" of target
  "mytarget_autogen_timestamp_deps" does not exist.
  ```

Even when converting the function to a macro (which has its own caveats anyway) the listing will produce the same error. Will have to think about this one a bit more.

--- a/android_openssl.cmake
+++ b/android_openssl.cmake
@@ -6,21 +6,26 @@ function(add_android_openssl_libraries)
   endif()
 
   if(Qt6_VERSION VERSION_GREATER_EQUAL 6.5.0)
-    if(NOT OPENSSL_ROOT_DIR)
-      set(OPENSSL_ROOT_DIR ${SSL_ROOT_PATH}/ssl_3/${CMAKE_ANDROID_ARCH_ABI})
-    endif()
-    list(APPEND android_extra_libs
-         ${ssl_root_path}/ssl_3/${CMAKE_ANDROID_ARCH_ABI}/libcrypto_3.so
-         ${ssl_root_path}/ssl_3/${CMAKE_ANDROID_ARCH_ABI}/libssl_3.so)
+    set(OPENSSL_CRYPTO_LIBRARY ${ssl_root_path}/ssl_3/${CMAKE_ANDROID_ARCH_ABI}/libcrypto_3.so)
+    set(OPENSSL_SSL_LIBRARY ${ssl_root_path}/ssl_3/${CMAKE_ANDROID_ARCH_ABI}/libssl_3.so)
+    set(OPENSSL_INCLUDE_DIR ${ssl_root_path}/ssl_3/include)
   else()
-    if(NOT OPENSSL_ROOT_DIR)
-      set(OPENSSL_ROOT_DIR ${SSL_ROOT_PATH}/ssl_1.1/${CMAKE_ANDROID_ARCH_ABI})
-    endif()
-    list(APPEND android_extra_libs
-         ${ssl_root_path}/ssl_1.1/${CMAKE_ANDROID_ARCH_ABI}/libcrypto_1_1.so
-         ${ssl_root_path}/ssl_1.1/${CMAKE_ANDROID_ARCH_ABI}/libssl_1_1.so)
+    set(OPENSSL_CRYPTO_LIBRARY ${ssl_root_path}/ssl_1.1/${CMAKE_ANDROID_ARCH_ABI}/libcrypto_1_1.so)
+    set(OPENSSL_SSL_LIBRARY ${ssl_root_path}/ssl_1.1/${CMAKE_ANDROID_ARCH_ABI}/libssl_1_1.so)
+    set(OPENSSL_INCLUDE_DIR ${ssl_root_path}/ssl_1.1/include)
   endif()
 
-  set_target_properties(${ARGN} PROPERTIES QT_ANDROID_EXTRA_LIBS
-                                           "${android_extra_libs}")
+  find_package(OpenSSL REQUIRED GLOBAL)
+  foreach(TARGET ${ARGN})
+    if(TARGET ${TARGET})
+      set_property(
+        TARGET ${TARGET}
+        APPEND
+        PROPERTY QT_ANDROID_EXTRA_LIBS ${OPENSSL_CRYPTO_LIBRARY} ${OPENSSL_SSL_LIBRARY}
+      )
+      target_link_libraries(${TARGET} PUBLIC OpenSSL::SSL OpenSSL::Crypto)
+    else()
+      message(WARNING "Invoked add_android_openssl_libraries on a non-existing target (${TARGET}), ignoring.")
+    endif()
+  endforeach()
 endfunction()

